db/migrate/20161231115932_add_full_text_search_indexes.rb

```diff
@@ -0,0 +1,9 @@
+class AddFullTextSearchIndexes < ActiveRecord::Migration[5.0]
+  def up
+    execute "create index notifications_subject_title on notifications using gin(to_tsvector('english', subject_title))"
+  end
+
+  def down
+    execute "drop index notifications_subject_title"
+  end
+end
```

Comment:
so this is a gin index for the 'subject_title' field, I ran an explain analyse on a text search query however it didn't seem to use this index 😬 maybe its because I need to vacuum analyse or maybe its not efficient to use this index on my small local dataset. However if we wanted to increase the number of fields to be included in the text search we will need to create another field to hold tokens of all the fields we want to search across.